pkg/tests/api/alerting/api_alertmanager_test.go

```diff
@@ -230,24 +251,71 @@ func TestAlertRuleCRUD(t *testing.T) {
                   "namespace_uid":"nsuid",
                   "namespace_id":1,
                   "rule_group":"arulegroup",
-                  "no_data_state":"",
-                  "exec_err_state":""
+                  "no_data_state":"Alerting",
+                  "exec_err_state":"KeepLastState"
                }
             }
          ]
       }
    ]
-}`, rulesNamespaceWithoutVariableValues(t, b))
+}`, rulesNamespaceWithoutVariableValues(t, b, ruleUIDs))
+		require.Equal(t, 2, len(ruleUIDs))
+		assert.NotEqual(t, ruleUIDs[0], ruleUIDs[1])
 	}
 
-	client := &http.Client{}
-	// Finally, make sure we can delete it.
+	// update the first rule and completely remove the other
```

Comment:
Are we? This reads as if we're updating one rule and deleting another, but we're actually updating one rule and then deleting the whole rule group.

Reply:
I mean that the result of this POST request will be to update the 1st rule created in [before](https://github.com/grafana/grafana/blob/3ae4df2cfe1dcbdd1f87daea666e7d66339007da/pkg/tests/api/alerting/api_alertmanager_test.go#L147) and delete the 2nd one because is not included in the payload.

Reply:
Makes sense, I'd include that as part of the description to make it clear. I was surprised given we include the ID of the rule in the payload... so I thought it would only update what we included an ID for.